### DISCUSSION TITLE
Examples for adding 3D elements to 2D games?

### DISCUSSION THREAD

**Discussion by c-robinson** [SolarLune/tetra3d#11] [General]
I ran across this library because I'd really like to add some 3D elements to a 2D game I'm thinking about, but honestly I have no idea where to get started and would love if there were some examples of doing so. My reason for asking is that one of the stated intents of the library (_"[f]or example, if you were making a 2D game with Ebiten but wanted to display just a few GUI elements or objects in 3D"_) is not present in the example set, which is almost entirely concerned with pure 3D applications of the library.

**Comment by SolarLune** [comment]
Hello!

It should be relatively simple - just like how the fully 3D example shows, you draw the camera's render result to the screen using an ebiten.Image, so the basic idea is to simply:

1) Place your 3D object in a scene.
2) Place your camera.
3) Render the scene using the camera.
4) Draw that render onscreen using the camera's texture (which is just an *ebiten.Image).

Whether you're making a 3D world where you move the camera around or simply leaving the camera stationary and moving and resizing the render result to match a 2D world doesn't really matter - the core principle is the same.

I hadn't considered making an example specifically for adding 3D elements to a 2D game, but I could do that.
